app/src/components/market/common/history_section/history_select.tsx

```diff
@@ -0,0 +1,354 @@
+`
+
+export const History_select: React.FC<Props> = ({
+  currency,
+  decimals,
+  fee,
+  holdingSeries,
+  isScalar,
+  marketMakerAddress,
+  onChange,
+  options,
+  outcomes,
+  scalarHigh,
+  scalarLow,
+  unit,
+  value,
+}) => {
+  const context = useConnectedWeb3Context()
+  const { library: provider } = context
+  const contracts = useContracts(context)
+  const { buildMarketMaker } = contracts
+  const marketMaker = buildMarketMaker(marketMakerAddress)
+  const [sharesData, setSharesData] = useState<FpmmTradeDataType[]>([])
+  const [sharesDataLoader, setSharesDataLoader] = useState<boolean>(true)
+
+  const outcomeArray: string[] = outcomes.length ? outcomes : ['Short', 'Long']
+  const data =
+    holdingSeries &&
+    holdingSeries
+      .filter(h => !!h.block)
+      .sort((a, b) => a.block.timestamp - b.block.timestamp)
+      .map(h => {
+        const prices = calcPrice(h.holdings.map(bigNumberify))
+        const outcomesPrices: { [outcomeName: string]: number } = {}
+        outcomeArray.forEach((k, i) => (outcomesPrices[k] = prices[i]))
+        return { ...outcomesPrices, date: formatTimestampToDate(h.block.timestamp, value) }
+      })
+  const [toggleSelect, setToggleSelect] = useState(true)
+  const [type, setType] = useState<HistoryType>(HistoryType.All)
+  const DropdownItems = [
+    {
+      content: 'All',
+      onClick: () => {
+        setType(HistoryType.All)
+      },
+    },
+    {
+      content: 'Liquidity',
+      onClick: () => {
+        setType(HistoryType.Liquidity)
+      },
+    },
+    {
+      content: 'Trades',
+      onClick: () => {
+        setType(HistoryType.Trades)
+      },
+    },
+  ]
+  const [pageIndex, setPageIndex] = useState(0)
+  const [pageSize] = useState(6)
+  const marketFeeWithTwoDecimals = Number(formatBigNumber(fee, 18))
+  const { fpmmTransactions, paginationNext, refetch, status } = useGraphFpmmTransactionsFromQuestion(
+    marketMakerAddress,
+    pageSize,
+    pageIndex,
+    type,
+    decimals,
+  )
+
+  useEffect(() => {
+    setSharesDataLoader(true)
+    ;(async () => {
+      if (fpmmTransactions) {
+        const cpk = await CPKService.create(provider)
+        const response: any[] = await Promise.all(
+          fpmmTransactions.map(async item => {
+            if (item.fpmmType === 'Liquidity') {
+              const block: any = await marketMaker.getTransaction(item.transactionHash)
+
+              return {
+                blockNumber: block,
+                id: item.id,
+                amount: item.collateralTokenAmount,
+                fpmmType: item.fpmmType,
+                poolShares: await marketMaker.poolSharesTotalSupplyByBlockNumber(block.blockNumber),
+                balances: await marketMaker.getBalanceInformationByBlock(
+                  marketMakerAddress,
+                  outcomes.length,
+                  block.blockNumber,
+                ),
```

Comment:
Shouldn't we already have access to these values?

Reply:
we don't have them in there actually....